firestore-send-email/functions/src/index.ts

```diff
@@ -86,14 +86,16 @@ function validateFieldArray(field: string, array?: string[]) {
 }
 
 async function processCreate(snap: FirebaseFirestore.DocumentSnapshot) {
-  return snap.ref.update({
-    delivery: {
-      startTime: admin.firestore.FieldValue.serverTimestamp(),
-      state: "PENDING",
-      attempts: 0,
-      error: null,
-    },
-  });
+  return admin.firestore().runTransaction((async transaction => {
```

Comment:
Note sure if you have a specific style that you are following, but in general it seems like your functions don't need to be marked async.

Reply:
This was to ensure the transaction returns a Promise. I'll change them to resolve a Promise from the callback instead.